java/src/main/native/ai_onnxruntime_OnnxTensor.c

```diff
@@ -101,32 +103,28 @@
  */
 JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OnnxTensor_createString
         (JNIEnv * jniEnv, jclass jobj, jlong apiHandle, jlong allocatorHandle, jstring input) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    OrtAllocator* allocator = (OrtAllocator*) allocatorHandle;
+  (void) jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*) apiHandle;
+  OrtAllocator* allocator = (OrtAllocator*) allocatorHandle;
 
-    // Extract the shape information
-    int64_t* shapeArr;
-    checkOrtStatus(jniEnv,api,api->AllocatorAlloc(allocator,sizeof(int64_t),(void**)&shapeArr));
-    shapeArr[0] = 1;
-
-    // Create the OrtValue
-    OrtValue* ortValue;
-    checkOrtStatus(jniEnv, api, api->CreateTensorAsOrtValue(allocator,shapeArr,0,ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING,&ortValue));
-
-    // Release the shape
-    checkOrtStatus(jniEnv, api, api->AllocatorFree(allocator,shapeArr));
+  // Create the OrtValue
+  int64_t shapeArr = 1;
+  OrtValue* ortValue;
```

Comment:
>OrtValue* ortValue; [](http://example.com/codeflow?start=2&length=19)

The API does not guarantee that ortValue would be null pointer on failure. It is better to initialize ahead of time.

Reply:
Will do. If it's non-null should it be released on failure? I'd assumed that any of the out parameters are left alone when an error code is returned.

Reply:
I think OrtValue should be ignored on a failure. The caller assumes responsibility only on success, otherwise it does not know what they are getting if anything.

Reply:
Ok, I'll set it to NULL and that should be sufficient then.